File: codechefquestion.py
t = int(input())

while t > 0:
    t-=1
    n,b = map(int,input().split())
    a = list(map(int,input().split()))
    l = []
    for i in a:
        if(i&b == b):
            l.append(i)
    if len(l) == 0:
     print("NO")
    else:
       x=l[0]
       for i in l:
          x = x&i
       if(x==b):
          print("YES")
       else:
          print("NO")

File: codechefquestion.cpp
#include<iostream>
#include<vector>

using namespace std;

int main(){

    int t;
    cin>>t;

    while(t--){
   int n,b;
   cin>>n>>b;
   vector<int> a(n);

   for(int i = 0;i<n;i++){

    cin>>a[i];
   }
   bool found =  false;
   for(int mask = 0;mask <(1<<n);mask++){
    int and_val = -1;

    for(int i = 0;i < n;i++){

        if(mask & (1<<i)){
            if(and_val == -1){
                and_val = a[i];
            }
            else{
                and_val &= a[i];
            }
        }
    }

if(and_val == b){
    found = true;
    break;
}


   }

   if(found){
    cout<<"YES"<<endl;
   }
   else{
    cout<<"NO"<<endl;
   }



    }
    return 0;;
}